ManagerRegistry.php

```diff
@@ -20,11 +20,40 @@
 
 namespace Doctrine\Bundle\PHPCRBundle;
 
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Bridge\Doctrine\ManagerRegistry as BaseManagerRegistry;
 use Doctrine\ODM\PHPCR\PHPCRException;
 
 class ManagerRegistry extends BaseManagerRegistry
 {
+    public function __construct(
+        ContainerInterface $container,
+        array $connections,
+        array $entityManagers,
+        $defaultConnectionName,
+        $defaultEntityManagerName,
+        $proxyInterfaceName
+    ) {
+        $parentTraits = class_uses(parent::class);
+        if (isset($parentTraits[ContainerAwareTrait::class])) {
+            // this case should be removed when Symfony 3.4 becomes the lowest supported version
+            // and then also, the constructor should type-hint Psr\Container\ContainerInterface
+            $this->setContainer($container);
+        } else {
+            $this->container = $container;
```

Comment:
does the new version still have the container property? but not accept the container in the constructor?

Reply:
this is adopted from the orm bundle. In on case there is the property in the abstract class in the other one it lives in the trait.